### PR TITLE
Added types for ICruiseResult

### PR DESCRIPTION
**Pull request by tharun208**
> Plan to do something drastic?  
> Leave an [issue](https://github.com/sverweij/dependency-cruiser/issues/new) with a
> summary of the changes you propose + some context on why you'd want to
> do that.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which fixes an issue without changing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The code I add will be subject to [The MIT license](../LICENSE), and I'm OK with that.
- [x] The code I've added is my own original work.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](./CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: types/dependency-cruiser.d.ts [hunk 1]
```diff
@@ -441,7 +879,7 @@ export interface IReporterOutput {
  *                        [here](../src/cli/flattenTypeScriptConfig.js))
  */
 export function cruise(
-  pFileDirArray: string[],
+  pFileDirArray: ICruiseResult[],
```
Comment (sverweij) [inline types/dependency-cruiser.d.ts]: This should stay a `string[]` - it's the files 'n folders the consumer puts in. You might've confused this parameter with the one in the `format` function below :-)

**Comment by sverweij** [comment]
Hi @tharun208 thanks for this PR! 

At a glance it's looking good - I'll be reviewing it on the short term. You might've seen a comment or two already when you read this. 

One thing I do have to ask though: could you tick the checkbox in the Checklist that says you've read (& understood) the [contributing ](https://github.com/sverweij/dependency-cruiser/blob/develop/.github/CONTRIBUTING.md) document? It a.o. ensures we can release your contribution under MIT license.

File: types/dependency-cruiser.d.ts (hunk 1 above)
Comment (tharun208) [reply types/dependency-cruiser.d.ts]: Okay, I will get it fixed

File: types/dependency-cruiser.d.ts [hunk 2]
```diff
@@ -406,13 +406,451 @@ export interface ICruiseOptions {
...
+}
+
+/**
+ * How severe a violation of a rule is. The 'error' severity will make some reporters return
+ * a non-zero exit code, so if you want e.g. a build to stop when there's a rule violated:
+ * use that. The absence of the 'ignore' severity here is by design; ignored rules don't
+ * show up in the output.
+ *
+ * Severity to use when a dependency is not in the 'allowed' set of rules. Defaults to 'warn'
+ */
```
Comment (sverweij) [inline types/dependency-cruiser.d.ts]: this piece of jsdoc pertains to the `severity` in `IRuleSummaryType` - can you move it up there?

Comment (tharun208) [reply types/dependency-cruiser.d.ts]: Sorry, Cant able to get this comment!

Comment (sverweij) [reply types/dependency-cruiser.d.ts]: I think the jsdoc belongs just over the `severity` attribute, like so:

```javascript
export interface IRuleSummaryType {
   /**
    * The (short, eslint style) name of the violated rule. Typically something like
    * 'no-core-punycode' or 'no-outside-deps'.
    */
   name: string;
   /**
    * How severe a violation of a rule is. The 'error' severity will make some reporters return
    * a non-zero exit code, so if you want e.g. a build to stop when there's a rule violated:
    * use that. The absence of the 'ignore' severity here is by design; ignored rules don't
    * show up in the output.
    *
    * Severity to use when a dependency is not in the 'allowed' set of rules. Defaults to 'warn'
    */
   severity: SeverityType;
 }
```

Comment (tharun208) [reply types/dependency-cruiser.d.ts]: sure got it

**Comment by tharun208** [comment]
Hi @sverweij, I will give a fresh PR with all the issues fixed.

**Pull request closed without merging**